src/merge.rs

```diff
@@ -393,9 +446,16 @@ impl<'t> Merger<'t> {
                        remote_parent_node,
                        local_parent_node);
 
-                let merged_child_node = self.two_way_merge(local_child_node, remote_child_node)?;
+                let mut merged_child_node = self.two_way_merge(local_child_node,
+                                                               remote_child_node)?;
+                if remote_child_node.diverged() || merged_node.guid != remote_parent_node.guid {
```

Comment:
`.diverged() || guids_different` seems a very common pattern here and used more often then a `.diverged()` alone is. It seems that it might be easy to get this wrong - I wonder if there's scope to capture this using the diverged mechanism somehow?

(This is more a rhetorical question than a suggestion as I'm still getting my head around this code)